tests/balena-api-mock.ts

```diff
@@ -65,19 +177,20 @@ class BalenaAPIMock {
 
 	// User details are cached in the SDK
 	// so often we don't know if we can expect the whoami request
-	public expectOptionalWhoAmI() {
-		this.scope
-			.get('/user/v1/whoami')
-			.optionally()
-			.reply(200, {
-				id: 99999,
-				username: 'testuser',
-				email: 'testuser@test.com',
-			});
+	public expectWhoAmI(persist = false, optional = true) {
```

Comment:
Right, this is nicer! Perhaps in future we can add these options generically for all of these expectations in some way.